app/controllers/api/v1/results_controller.rb

```diff
@@ -80,6 +94,30 @@ def create_text_result
             result_text.save!
           end
         end
+        # rubocop:enable Metrics/BlockLength:
+      end
+
+      def update_text_result
+        raise NotImplementedError, 'update_text_result should be implemented!'
+      end
+
+      def create_file_result
+        Result.transaction do
+          @result = @task.results.create!(result_params.merge(user_id: current_user.id))
+          asset = Asset.create!(result_file_params)
+          ResultAsset.create!(asset: asset, result: @result)
+        end
+      end
+
+      def update_file_result
+        old_checksum, new_checksum = nil
+        Result.transaction do
+          old_checksum = @result.asset.file.blob.checksum
+          @result.asset.file.purge
```

Comment:
There is no need to delete file manually, ActiveStorage will do it automatically with delayed job